libraries/BLE/src/BLECharacteristic.cpp

```diff
@@ -480,10 +480,20 @@ void BLECharacteristic::notify(bool is_notification) {
 	assert(getService() != nullptr);
 	assert(getService()->getServer() != nullptr);
 
+	auto cb = BLECharacteristicCallbacks();
```

Comment:
There is no need to create additional callback, which will be created easch time you send notification/indications (memory leak).

Just use the same mechanism like with onRead/onWrite:
```
if(m_pCallback != nullptr){
     callback->onNotify(this);
}
```

Reply:
@chegewara I personally prefer to use [Null object pattern](https://en.wikipedia.org/wiki/Null_object_pattern) to maintain [KISS](https://en.wikipedia.org/wiki/KISS_principle) and [DRY](https://ru.wikipedia.org/wiki/Don’t_repeat_yourself)

It's not a big deal for performance or memory considerations with BLE data rates. But much simpler to read, isn't it?

Reply:
If it wont make memory leak, then ok.

Reply:
@chegewara It's a scoped variable, placed on stack. It is automatically deleted when we leave its outbound `{}` block ( see [RAII](https://en.wikipedia.org/wiki/Resource_acquisition_is_initialization) )

Reply:
for consistency it should be using the same approach as onRead/onWrite...

also this shouldn't leak any memory as it is allocated and freed automatically as part of the method execution.

Reply:
@atanisoft Did I got you right: you mean use Null object pattern in other callback checks?

Reply:
The general approach used so far (in this and other libs) is:
```
if(m_pCallback){
     m_pCallback->onNotify(this);
}
```
I'm not certain your approach is any cleaner/readable.

Reply:
@atanisoft so, you mean readability of the code with extra

```
if(m_pCallback){
     m_pCallback->onNotify(this);
}
```

in dozen places with other 5-level conditionals is much cleaner than  
```
callback->onNotify(this);
```

and we should use that approach because we used to? This doesn't sound convictive to me, sorry. I'd prefer to replace other usages of that callback to decrease overall codebase and improve maintainability.

Reply:
I'm fine either way, just be consistent in usage :)

Reply:
@atanisoft Thanks, Mike